tools/iscc-sum/iscc_similarity.xml

```diff
@@ -0,0 +1,211 @@
+            echo \$ISCC1 > temp1.txt &&
+            echo \$ISCC2 > temp2.txt &&
+
+            ## Use iscc-sum to compare similarity and output directly
+            iscc-sum --similar --threshold $threshold temp1.txt temp2.txt > '${output_file}' || true
+        #else:
+            ## Single collection - find similar files within
+            mkdir -p input_files &&
+            #for $file in $input_type.file_collection:
+                ln -s '$file' 'input_files/$file.element_identifier' &&
+            #end for
+            iscc-sum --similar --threshold $threshold input_files/* > '${output_file}'
+        #end if
+    ]]></command>
+
+    <inputs>
+        <conditional name="input_type">
+            <param name="input_selector" type="select" label="Input type">
+                <option value="two_files">Compare two files</option>
+                <option value="collection">Find similar files in collection</option>
+                <option value="two_collections">Compare two collections</option>
+            </param>
+            <when value="two_files">
+                <param name="file1" type="data" format="data" label="First file"/>
+                <param name="file2" type="data" format="data" label="Second file"/>
+            </when>
+            <when value="collection">
+                <param name="file_collection" type="data_collection" collection_type="list"
+                       format="data" label="File collection to compare"/>
+            </when>
+            <when value="two_collections">
+                <param name="collection1" type="data_collection" collection_type="list" format="data"
+                       label="First collection (reference)"
+                       help="Reference collection - will generate combined ISCC"/>
+                <param name="collection2" type="data_collection" collection_type="list" format="data"
+                       label="Second collection (to compare)"
+                       help="Collection to compare against reference - will generate combined ISCC"/>
+            </when>
+        </conditional>
+        <param name="threshold" type="integer" value="12" min="0" max="256"
+               label="Similarity threshold (Hamming distance)"
+               help="Maximum Hamming distance for similarity matching. 0-5: Nearly identical, 6-12: Likely similar (default), 13-20: Probably somewhat similar"/>
+    </inputs>
+    
+    <outputs>
+        <data name="output_file" format="txt" label="${tool.name} on ${on_string}"/>
+    </outputs>
+    
+    <tests>
+        <!-- Test 1: Pairwise file comparison -->
+        <test expect_num_outputs="1">
+            <conditional name="input_type">
+                <param name="input_selector" value="two_files"/>
+                <param name="file1" value="test1.png"/>
+                <param name="file2" value="test1.png"/>
+            </conditional>
+            <param name="threshold" value="12"/>
```

Comment:
For tighter testing, wouldn't it make sense to set this threshold to 0, since the two inputs are identical?